packages/core/src/useStateGetters.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import { NavigatorStateGetter } from './NavigationBuilderContext';
+
+/**
+ * Hook which lets child navigators add getters to be called for obtaining stage.
+ */
+export default function useStateGetters() {
+  const stateGetters = React.useRef<Record<string, NavigatorStateGetter>>({});
+
+  const addStateGetter = React.useCallback(
+    (getter: NavigatorStateGetter, key: string) => {
+      stateGetters.current[key] = getter;
+
+      return () => {
+        delete stateGetters.current[key];
+      };
+    },
+    [stateGetters]
```

Comment:
This shouldn't need to be in the deps array.
```suggestion
    []
```